### ISSUE TITLE
Step ADD function in the Step Sequencer 

### ISSUE THREAD

**Issue by CaBachhav** [LMMS/lmms#4823]
this is the function we see in oldskool Amiga and Fasttracker . If we implant this to our beloved LMMS step sequencer , believe me friends, LMMS will become better than FLStuio.  step ADD can be anything from 1 to 32 .

**Comment by CaBachhav** [comment]
due to the love with LMMS only , Im learning c,c++ .

**Comment by zonkmachine** [comment]
Thank you for your suggestion and interest in lmms. Can you please explain more what this is about and just presume that people haven't used these applications before. How does this differ from how the step editor adds notes today (Choose note length and add note by pressing a key or use arrow keys to move)?

If you have questions, please bring them up on our Discord site.
https://discord.gg/3sc5su7
Especially the channels `#support` and `#techtalk

**Comment by CaBachhav** [comment]
Today's step sequencer jumps to the next empty step when an input is given by pc keys or midi keyboard.this is ok. What we want is a ADD button should be made near step recorder . The ADD can be Defined by the user anything between 1 to 32 step. Ex. If we defined ADD to be 4. Then on each input pc key stroke the CURSER jumps to next 4th step. This function is very very useful as it boost the speed of sequencing beats and melodies. This function is there in oldskool AMIGA mod trackers and Protraker .we want to implement it in LMMS  piano roll sequencer.

**Comment by musikBear** [comment]
**Song-editor**? Is that what you call "step sequencer"? -Because **Beat-editor** already has features to add steps 
Besides the 'conversation driven' Discord-site, where it is difficult to see if a questions is answered or not, we also have a real **Forum** where questions are in easy to follow threads, like here: https://lmms.io/forum/viewforum.php?f=7

**Comment by zonkmachine** [comment]
> **Song-editor**? Is that what you call "step sequencer"? -Because **Beat-editor** already has features to add steps

No. He is talking about the Step Recording feature in the Piano Roll.

**Comment by CaBachhav** [comment]
I am suggesting a CURSOR JUMP function to be added in piano roll step sequencer. Plz see the ADD function in fast tracker on Pro tracker or OpenMpt.

**Comment by zonkmachine** [comment]
> Plz see the ADD function in fast tracker on Pro tracker or OpenMpt.

Can you please instead link to information about this function?

**Comment by musikBear** [comment]
> No. He is talking about the Step Recording feature in the Piano Roll.

Holy Cr.. I have no idea of what that is 8| 
Are you referring to either of the two "record" options
...or is there actually a feature i dont know after 12 years...... :0+
...ASHhhh :/
Isent he just looking for at way to push an opening into existing notes - Eg "Add empty steps" into an existing sequence of notes???
@CaBachhav LMMS have that, but its a bit 'hidden'
**IF** thats what you mean....
I can show you in a video, but if i'm once again wayyyyy-off, no need for that....

Place play-head where you want to add empty-steps, and then hold SH and tap INSERT -every press adds a

**Comment by musikBear** [comment]
No. He is talking about the Step Recording feature in the Piano Roll.
Holy Cr.. I have no idea of what that is 8|
Are you referring to either of the two "record" options
...or is there actually a feature i dont know after 12 years...... :0+
...ASHhhh :/
Isent he just looking for at way to push an opening into existing notes - Eg "Add empty steps" into an existing sequence of notes???
@CaBachhav LMMS have that, but its a bit 'hidden'
IF thats what you mean....
I can show you in a video, but if i'm once again wayyyyy-off, no need for that....

**Comment by musikBear** [comment]
>No. He is talking about the Step Recording feature in the Piano Roll.
Holy Cr.. I have no idea of what that is 8|
Are you referring to either of the two "record" options
...or is there actually a feature i dont know after 12 years...... :0+
...ASHhhh :/
Isent he just looking for at way to push an opening into existing notes - Eg "Add empty steps" into an existing sequence of notes???
@CaBachhav LMMS have that, but its a bit 'hidden'
IF thats what you mean....
I can show you in a video, but if i'm once again wayyyyy-off, no need for that....

**Comment by musikBear** [comment]
>No. He is talking about the Step Recording feature in the Piano Roll.

Holy Cr.. I have no idea of what that is 8|
Are you referring to either of the two "record" options
...or is there actually a feature i dont know after 12 years...... :0+
...ASHhhh :/
Isent he just looking for at way to push an opening into existing notes - Eg "Add empty steps" into an existing sequence of notes???
@CaBachhav LMMS have that, but its a bit 'hidden'
IF thats what you mean....
I can show you in a video, but if i'm once again wayyyyy-off, no need for that....

**Comment by CaBachhav** [comment]
Some programmer are adding Step Sequencer style function in PIANO ROLL EDITOR . If you know this , then I suggested a function to add in this step sequencer like ex. I don't want to use mouse clicks to program drums or simple melodies , I can put it using pc keys , but I want to add note on 1st , 5th, 9th,13th beat of each 16 beat bar . If I put a note on 1st beat , CURSER goes to 2nd , then 3rd going +1. I WANT TO CUSTOMISE IT FROM + 1 TO +32 . So that if I set it to +4 mode. cursor will automatically jump to 5th, then 9th , then 13 th , then 17th and forth. This is what I wanted .
